src/components/Dropdown/Dropdown.tsx

```diff
@@ -222,48 +229,96 @@ export default class Dropdown extends AutoControlledComponent<
+            const { innerRef, ...accessibilityRootPropsRest } = accessibilityRootProps
             return (
-              <div
-                className={classes.containerDiv}
-                onClick={this.handleContainerClick.bind(this, isOpen)}
-              >
-                {multiple && this.renderSelectedItems(styles)}
-                {search &&
-                  this.renderSearchInput(
-                    getRootProps,
-                    getInputProps,
+              <Ref innerRef={innerRef}>
+                <div
+                  className={classes.container}
+                  onClick={multiple ? this.handleContainerClick.bind(this, isOpen) : undefined}
+                >
+                  {multiple && this.renderSelectedItems(styles)}
+                  {search
+                    ? this.renderSearchInput(
+                        accessibilityRootPropsRest,
+                        getInputProps,
+                        highlightedIndex,
+                        selectItemAtIndex,
+                        variables,
+                      )
+                    : this.renderTriggerButton(styles, getToggleButtonProps)}
+                  {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
+                  {this.renderItemsList(
+                    styles,
+                    variables,
+                    isOpen,
                     highlightedIndex,
+                    toggleMenu,
                     selectItemAtIndex,
+                    getMenuProps,
+                    getItemProps,
+                    getInputProps,
                   )}
-                {toggleButton && this.renderToggleButton(getToggleButtonProps, styles, isOpen)}
-                {this.renderItemsList(
-                  styles,
-                  variables,
-                  getMenuProps,
-                  getItemProps,
-                  isOpen,
-                  highlightedIndex,
-                )}
-              </div>
+                </div>
+              </Ref>
             )
           }}
         </Downshift>
       </ElementType>
     )
   }
 
+  private renderTriggerButton(
+    styles: ComponentSlotStylesInput,
+    getToggleButtonProps: (options?: GetToggleButtonPropsOptions) => any,
+  ): JSX.Element {
+    const { placeholder, itemToString, multiple } = this.props
+    const { value } = this.state
+    let content
+
+    if (multiple) {
+      content = placeholder
+    } else {
+      content = value ? itemToString(value) : placeholder
+    }
+
+    return (
+      <Ref innerRef={this.buttonRef}>
+        <Button
+          content={content}
+          fluid
+          styles={styles.button}
+          {...getToggleButtonProps({
+            onFocus: () => {
+              this.setState({ focused: true })
+            },
+            onBlur: () => {
+              this.setState({ focused: false })
+            },
+            'aria-label': content, // TODO: add this to behaviour
```

Comment:
It seems for me that ```aria-label``` is redundant here. We pass the same content to Button, screen readers can cope with that and read content inside a button element.

Reply:
it wasn't redundant, I used it to override the one from Downshift's `getToggleButtonProps`. will leave it there :)